src/components/CustomCollection/CustomBookmark/CustomBookmarkForm.test.tsx

```diff
@@ -0,0 +1,241 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import {
+  render,
+  screen,
+  RenderResult,
+  act,
+  waitFor,
+} from '@testing-library/react'
+import React, { createRef } from 'react'
+import { axe } from 'jest-axe'
+import { CustomBookmarkForm } from './CustomBookmarkForm'
+import userEvent from '@testing-library/user-event'
+
+describe('CustomBookmarkForm component', () => {
+  const testHandlers = {
+    onSave: jest.fn(),
+    onCancel: jest.fn(),
+  }
+
+  let html: RenderResult
+
+  beforeEach(async () => {
+    const mockNameRef = createRef<HTMLInputElement>()
+    const mockUrlRef = createRef<HTMLInputElement>()
+
+    html = render(
+      <CustomBookmarkForm
+        onSave={testHandlers.onSave}
+        onCancel={testHandlers.onCancel}
+        nameInputRef={mockNameRef}
+        urlInputRef={mockUrlRef}
+      />
+    )
+    await screen.findByLabelText('Name')
+  })
+
+  it('renders the CustomBookmarkForm component', async () => {
+    expect(screen.getByLabelText('Name')).toBeInTheDocument()
+    expect(screen.getByLabelText('URL')).toBeInTheDocument()
+    expect(
+      screen.getByRole('button', { name: 'Save custom link' })
+    ).toBeInTheDocument()
+  })
+
+  it('validates the bookmarkLabel field', async () => {
+    // Input: On load, empty
+    // Result: Invalid
+    const labelInput = screen.getByLabelText('Name')
+    expect(labelInput).toBeInvalid()
+
+    // Input: String
+    // Result: Valid
+    userEvent.type(labelInput, 'My Custom Link')
+    await waitFor(() => expect(labelInput).toBeValid())
```

Comment:
praise: I'm glad you worked this out it's much better than randomly adding `act` imo.

Reply:
thank you!! yes it feels *much* better